crossdc-producer/src/test/java/org/apache/solr/crossdc/SolrAndKafkaIntegrationTest.java

```diff
@@ -64,19 +62,24 @@
   protected static volatile MiniSolrCloudCluster solrCluster1;
   protected static volatile MiniSolrCloudCluster solrCluster2;
 
-  protected static volatile Consumer consumer = new Consumer();
+  protected static volatile Consumer consumer;
 
   private static String TOPIC = "topic1";
 
   private static String COLLECTION = "collection1";
   private static String ALT_COLLECTION = "collection2";
+  private static Thread.UncaughtExceptionHandler uceh;
 
   @BeforeClass
```

Comment:
[-1/Q] Each test case in this class relies pretty heavily on the assumption that when the test-method starts, COLLECTION will have 0 docs on both primary and secondary.

Won't moving the Solr cluster creation/deletion logic from Before/After to BeforeClass/Afterclass break that assumption everywhere?  Or am I overlooking somewhere that deletes-all-docs or deletes/recreates collections in between test methods?

Reply:
I've switched back and forth a couple times trying to figure a fail I get with testMirroringUpdateProcessor, so it may have been off in the iteration that was up, I'm not sure, that latest is back to beforeclass, afterclass because I moved the test method that prompted that change initially into its own test class anyway.

The issue I've been seeing is unrelated though - the doc that goes missing in the secondary dc is dealing with collection2 and does get cleaned up in the test method itself, and it also is an issue even when just run on it's own for me.